crates/rspack_core/src/normal_module.rs

```diff
@@ -466,7 +464,14 @@ impl NormalModule {
 impl Identifiable for NormalModule {
   #[inline]
   fn identifier(&self) -> ModuleIdentifier {
-    self.id
+    if !matches!(
+      self.module_type,
+      ModuleType::Js | ModuleType::Ts | ModuleType::Tsx
+    ) {
+      ModuleIdentifier::from(self.request.as_ref())
+    } else {
+      ModuleIdentifier::from(format!("{}|{}", self.module_type, self.request))
```

Comment:
This might have a performance issue, every time you call `module.identifier` module identifier will be recalculated